### DISCUSSION TITLE
I pulled a VoucherVision update and now it's not working. What do I do?

### DISCUSSION THREAD

**Discussion by Gene-Weaver** [Gene-Weaver/VoucherVision#11] [Q&A / FAQs]
1. Make sure that you check README.md for any new instructions
2. Upgrade packages installed in your virtual environment. We use several packages that are updated every week and occasionally that causes issues. For example, LangChain frequently refactors its codebase and requires renamed packages. Also, API workflows change frequently so OpenAI, Mistral, Google, LangChain packages should be upgrades.
3. If step 2 fails, rebuild your virtual environment from scratch.
4. Contact the developers or look in [Issues](https://github.com/Gene-Weaver/VoucherVision/issues) for help.

**Comment by mickley** [comment]
From experience, here are a few other things that might be useful when upgrading:
- Rename your virtual environment and start a new one, installing everything from scratch. This allows you to go back to the old environment if needed, but in most cases it's safer to just start with a clean environment tailored to the latest Vouchervision version. 
- Rename PRIVATE_DATA.yaml, which stores the API keys. If any APIs were added, the old file will break VoucherVision
- Rename expense_report/expense_report.csv. If any columns were added to the report, the old file will break Vouchervision
